dvc/repo/__init__.py

```diff
@@ -319,22 +320,30 @@ def used_cache(
         ):
             targets = targets or [None]
 
-            pairs = cat(
-                self.stage.collect_granular(
-                    target, recursive=recursive, with_deps=with_deps
+            try:
+                pairs = cat(
+                    self.stage.collect_granular(
+                        target, recursive=recursive, with_deps=with_deps
+                    )
+                    for target in targets
                 )
-                for target in targets
-            )
 
-            suffix = f"({branch})" if branch else ""
-            for stage, filter_info in pairs:
-                used_cache = stage.get_used_cache(
-                    remote=remote,
-                    force=force,
-                    jobs=jobs,
-                    filter_info=filter_info,
-                )
-                cache.update(used_cache, suffix=suffix)
+                suffix = f"({branch})" if branch else ""
+                for stage, filter_info in pairs:
+                    used_cache = stage.get_used_cache(
+                        remote=remote,
+                        force=force,
+                        jobs=jobs,
+                        filter_info=filter_info,
+                    )
+                    cache.update(used_cache, suffix=suffix)
+            except DvcException as e:
+                logger.error(f"Cache for '{branch}' could not be collected")
+                if ignore_dvc_exceptions:
```

Comment:
Seems like using `--force` would be intuitive here.
```suggestion
                if ignore_dvc_exceptions or force:
```

Reply:
Do you suggest using force and ignore_dvc_exceptions or just force?
Indeed, `--force` seems more coherent with options across dvc commands

Reply:
@courentin Seems like `--force` would be enough.

Reply:
Hum, `--force` is already used in the gc command "Force garbage collection - automatically agree to all prompts".

Reply:
@efiop Don't we want to differentiate `--force` where the behaviour is "Force garbage collection - automatically agree to all prompts" and `--skip-failing-collect` so that we do one without the other?

WDYT?

I'd like to merge this PR so that our team can clean their workspace :)

Reply:
@courentin Do you have a scenario in which those could be used separately?

Reply:
I'd say if we want to ignore failing commits but not agree to all prompts. But to be honest I'm fine with using only `--force`, I let you decide what's best for the product ;)